pkg/importer/datastream_test.go

```diff
@@ -43,38 +41,36 @@ var _ = Describe("Streaming Data Conversion", func() {
 
 		for _, t := range tests {
 
-			var testData []byte
+			var base, input []byte
 			desc := t.testDesc
 			fn := t.expectFileName
 			ff := t.expectFileFormat
 
 			BeforeEach(func() {
 				// Create a some random data to compress and/or archive.
-				b := make([]byte, size)
-				n, err := rand.Read(b)
-				Expect(n).To(Equal(len(b)), "The number of random bytes written does not match the buffer size")
+				base = make([]byte, size)
+				n, err := rand.Read(base)
+				Expect(n).To(Equal(len(base)), "The number of random bytes written does not match the buffer size")
 				Expect(err).NotTo(HaveOccurred(), "Error occurred during rand.Read()")
 
 				// Generate the expected data format from the random bytes
-				testData, err = f.FormatTestData(b, fn, ff)
+				input, err = f.FormatTestData(base, ff)
 				Expect(err).NotTo(HaveOccurred())
 			})
 
 			It(desc, func() {
-				buf := bytes.NewReader(testData)
-				r, err := image.UnpackData(fn, ioutil.NopCloser(buf))
+
+				inBuf := bytes.NewReader(input)
```

Comment:
I don't understand why the `BeforeEach` is inside the `for` loop. BeforeEach'es are triggered by each `It`. To me, defining the `BeforeEach` before the `for` loop makes more sense but maybe I'm missing something here?